Automattic/jetpack
# Protect: Update use fixers query error prop handling

Fixes #
The following two endpoint used in the process of triggering fixers and retrieving fixer statuses:
- `/site/rewind/threats/fix/batch/start`
- `/site/rewind/threats/fix/batch/status`

Can each return a variety of responses:
- Top level error `[ 'ok' => false, 'error' => string ]`
- Threat level error `[ 'ok' => true, 'threats' => [ <threat_id> => [ 'error' => string ] ] ]`
- Success `[ 'ok' => true, 'threats' => [ <threat_id> => [ 'status' => string, 'last_updated' => string ] ] ]` 

We only currently account for the success return type.

At any time in the process of fixing threats and retrieving statuses we will need account for these potentially responses and handle them accordingly.

## Proposed changes:

* Update our `FixerStatus` type and all the variable and methods that use it to account for the potential variations
* Account for and handle the potential response variations in the `useFixersMutations` and `useFixersQuery` hooks

### Other information:

## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/1422

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Checkout this branch
* Disable the CredentialsGate on the `fixerAllThreatsModal` and the `fixThreatsModal`
* Manually update `REST_Controller::api_fix_threats` to supply a top level error, threat level error, and success (of varying statuses)
* Manually update `REST_Controller::api_fix_threats_status` to supply a top level error, threat level error, and success (of varying statuses)
* Run your Jurassic Tube and connect your user
* Add fixable threats
* Apply various combinations above to verify that the situations this PR is intended to address are fixed
* Ensure no regressions are introduced
